Task1/todo.js

```diff
@@ -0,0 +1,134 @@
+    if(todo.deleted)
+    {
+        item.remove();
+        return
+    }
+
+    const node=document.createElement("li");
+    //create attribute as it uniquely define the key
+    node.setAttribute('todo-id',todo.id);
+    //creating delelte btn
+    const deleteButton =document.createElement("button");
+    deleteButton.classList.add('js-delete-todo');
+    deleteButton.innerHTML="Delete" 
+    //creating span 
+    node.innerHTML = `
+    <span>${todo.text}</span>
+    `;
+    //appending node and btn
+    node.appendChild(deleteButton)
+    list.appendChild(node)
+}
+
+//function to add item in an array
+function addTodo(text)
+{
+    const todo={
+        text,
+        id:Date.now(),
+    };
+
+    todoItems.push(todo);
+    //changed this 
+    todoItems.forEach((todo)=>{
+        renderTodo(todo);
+    });
+    // renderTodo(todo);
+}
+
+
+
+//function to delete each item
+function deleteTodo(idToBeRemoved)
+{
+
+    const index=todoItems.findIndex(({id})=>id===Number(idToBeRemoved));
+    //create an object 
+    const todo={
+        deleted:true,
+        ...todoItems[index]
+    };
+    //create an array after removing the items 
+    todoItems=todoItems.filter(({id})=>id!==Number(idToBeRemoved));
+    renderTodo(todo);
+}
+
+
+
+//Select the form element
+const form = document.querySelector("#todoForm");
+//Add event listener
+const add = document.querySelector("#button");
+add.addEventListener('click',event=>{
+    //prevent page from refresh on clicking
+    event.preventDefault();
+    //select the input
+    const input=document.querySelector("#todoInput");
+    //get value of input and remove whitespaces we can use createNodetext also
+    const text=input.value.trim();
+    //updated from text!=='' to text 
+    if(text)
+    {
+        addTodo(text);
+        //reset value after pushing to ''
+        input.value='';
+        //give focus to element
+        input.focus();
+    }
+});
+
+
+
+//selecting the delete button
+const list=document.querySelector('#todoList');
+list.addEventListener('click',event=>{
+    if(event.target.classList.contains('js-delete-todo'))
+    {
+        const itemKey=event.target.parentElement;
+        deleteTodo(itemKey.getAttribute('todo-id'));
+        //getattribute gives string in retrun
+    }
+});
+
+
+//function to delelte all items
+const deleteall=document.querySelector('.delete-all');
+deleteall.addEventListener('click',event=>{
+    while(todoItems.length>0)
+    {
+        todoItems.pop();//using pop as it is faster
+    }
+    //todoItem.length=0;
+    //todoItems=[];
+})
+
+
+//function to search
+// const searchBox = document.getElementById('searchBox');
+ const searchBox = document.getElementById('todoInput');
+
+//selecting search box
+searchBox.addEventListener('input',()=>{
+    //resetting the list 
+    list.innerHTML="";
+    const valueToBeSearched=searchBox.value.trim();
+    todoItems.forEach(item=>{
+        //destructuring the object 
+        const {text} = item;
```

Comment:
The object can be destructured in the callback itself. No need for a new variable. Example:

```
todoItems.forEach({ text }=>{
        //destructuring the object 
        if(text.includes(valueToBeSearched)){
            renderTodo(item);
        };
    })
```
@Poorvi-3pillar